diff --git a/CMakeLists.txt b/CMakeLists.txt
index f886cf2e86..7910ed6556 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -296,7 +296,9 @@ if(EXECUTORCH_BUILD_TESTS)
 endif()
 
 # TODO(dbort): Fix these warnings and remove this flag.
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(NOT MSVC)
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 # TODO(#6475): This requires/assumes that the repo lives in a directory named
@@ -786,7 +788,9 @@ if(EXECUTORCH_BUILD_PYBIND)
       bundled_module PUBLIC ${_common_include_directories}
     )
     target_compile_options(
-      bundled_module PUBLIC -Wno-deprecated-declarations -fPIC
+      bundled_module PUBLIC
+      $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -fPIC>
+      $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
     )
   endif()
 
@@ -858,9 +862,13 @@ if(EXECUTORCH_BUILD_PYBIND)
   endif()
 
   # compile options for pybind
-  set(_pybind_compile_options -Wno-deprecated-declarations -fPIC -frtti
-                              -fexceptions
-  )
+  if(MSVC)
+    set(_pybind_compile_options /wd4996 /GR /EHsc)
+  else()
+    set(_pybind_compile_options -Wno-deprecated-declarations -fPIC -frtti
+                                -fexceptions
+    )
+  endif()
 
   # util lib
   add_library(
diff --git a/backends/apple/coreml/CMakeLists.txt b/backends/apple/coreml/CMakeLists.txt
index 9879a05e3d..a0ad095ebc 100644
--- a/backends/apple/coreml/CMakeLists.txt
+++ b/backends/apple/coreml/CMakeLists.txt
@@ -113,7 +113,7 @@ target_include_directories(
 if(APPLE)
   target_link_libraries(coreml_util PRIVATE ${FOUNDATION_FRAMEWORK})
 endif()
-target_compile_options(coreml_util PUBLIC -fPIC)
+target_compile_options(coreml_util PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-fPIC>)
 
 install(TARGETS coreml_util DESTINATION lib)
 
@@ -150,7 +150,7 @@ if(APPLE)
     coreml_inmemoryfs PRIVATE coreml_util ${FOUNDATION_FRAMEWORK}
   )
 endif()
-target_compile_options(coreml_inmemoryfs PUBLIC -fPIC)
+target_compile_options(coreml_inmemoryfs PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-fPIC>)
 
 install(
   TARGETS coreml_inmemoryfs
@@ -178,7 +178,7 @@ if(EXECUTORCH_BUILD_PYBIND)
     executorchcoreml PRIVATE coreml_util coreml_inmemoryfs
                              nlohmann_json::nlohmann_json
   )
-  target_compile_options(executorchcoreml PUBLIC -fPIC)
+  target_compile_options(executorchcoreml PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-fPIC>)
 endif()
 
 # coremldelegate
@@ -242,8 +242,8 @@ if(APPLE)
 
   if(EXECUTORCH_BUILD_DEVTOOLS)
     target_compile_options(executorch_core PUBLIC -DET_EVENT_TRACER_ENABLED)
-    target_compile_options(coremldelegate PRIVATE "-frtti")
-    target_compile_options(libprotobuf-lite PRIVATE "-frtti")
+    target_compile_options(coremldelegate PRIVATE $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti> $<$<CXX_COMPILER_ID:MSVC>:/GR>)
+    target_compile_options(libprotobuf-lite PRIVATE $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti> $<$<CXX_COMPILER_ID:MSVC>:/GR>)
   else()
     target_compile_options(coremldelegate PRIVATE "-fno-rtti")
   endif()
diff --git a/backends/apple/mps/CMakeLists.txt b/backends/apple/mps/CMakeLists.txt
index 5a253347b0..757151bb4e 100644
--- a/backends/apple/mps/CMakeLists.txt
+++ b/backends/apple/mps/CMakeLists.txt
@@ -18,7 +18,11 @@ endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
-set(_common_compile_options -Wno-deprecated-declarations)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations)
+endif()
 
 set(_mps_schema__include_dir "${CMAKE_BINARY_DIR}/schema/include")
 
@@ -48,6 +52,7 @@ add_custom_command(
 
 add_library(mps_schema INTERFACE ${_mps_schema__outputs})
 set_target_properties(mps_schema PROPERTIES LINKER_LANGUAGE CXX)
+add_dependencies(mps_schema flatc)
 target_include_directories(
   mps_schema
   INTERFACE
diff --git a/backends/arm/CMakeLists.txt b/backends/arm/CMakeLists.txt
index ede7a96a38..5b65abf177 100644
--- a/backends/arm/CMakeLists.txt
+++ b/backends/arm/CMakeLists.txt
@@ -22,7 +22,9 @@ add_compile_definitions(C10_USING_CUSTOM_GENERATED_MACROS)
 # bare metal backend builds
 if(EXECUTORCH_BUILD_ARM_BAREMETAL)
 
-  add_compile_options("-Wall" "-Werror")
+  if(NOT MSVC)
+    add_compile_options("-Wall" "-Werror")
+  endif()
 
   # Third-party folder and Ethos-U driver inclued
   set(THIRD_PARTY_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/third-party")
diff --git a/backends/cadence/CMakeLists.txt b/backends/cadence/CMakeLists.txt
index 47183bed21..e73cf76af2 100644
--- a/backends/cadence/CMakeLists.txt
+++ b/backends/cadence/CMakeLists.txt
@@ -31,7 +31,11 @@ add_compile_definitions(C10_USING_CUSTOM_GENERATED_MACROS)
 if(EXECUTORCH_CADENCE_CPU_RUNNER)
   include(${EXECUTORCH_ROOT}/tools/cmake/Codegen.cmake)
 
-  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+  if(MSVC)
+    set(_common_compile_options /wd4996)
+  else()
+    set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+  endif()
 
   # Find prebuilt libraries. executorch package should contain portable_ops_lib,
   # etdump, bundled_program.
diff --git a/backends/mediatek/CMakeLists.txt b/backends/mediatek/CMakeLists.txt
index ed9b37e199..a24aa3680e 100644
--- a/backends/mediatek/CMakeLists.txt
+++ b/backends/mediatek/CMakeLists.txt
@@ -23,7 +23,10 @@ include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/runtime/include)
 
 # targets
 add_library(neuron_backend SHARED)
-target_compile_options(neuron_backend PRIVATE "-frtti" "-fexceptions")
+target_compile_options(neuron_backend PRIVATE
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti -fexceptions>
+  $<$<CXX_COMPILER_ID:MSVC>:/GR /EHsc>
+)
 target_link_libraries(
   neuron_backend PRIVATE executorch_core portable_ops_lib portable_kernels
                          android log
diff --git a/backends/openvino/CMakeLists.txt b/backends/openvino/CMakeLists.txt
index 4d32d8932c..5f86624902 100644
--- a/backends/openvino/CMakeLists.txt
+++ b/backends/openvino/CMakeLists.txt
@@ -35,7 +35,10 @@ find_package(OpenVINO REQUIRED)
 add_library(openvino_backend STATIC .)
 
 # Enable exceptions and RTTI for OpenVINO backend
-target_compile_options(openvino_backend PRIVATE -frtti -fexceptions)
+target_compile_options(openvino_backend PRIVATE
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti -fexceptions>
+  $<$<CXX_COMPILER_ID:MSVC>:/GR /EHsc>
+)
 
 # Include Executorch directories
 target_include_directories(openvino_backend PRIVATE ${COMMON_INCLUDE_DIRS})
diff --git a/backends/qualcomm/CMakeLists.txt b/backends/qualcomm/CMakeLists.txt
index 3210559726..cb3fef7ab1 100644
--- a/backends/qualcomm/CMakeLists.txt
+++ b/backends/qualcomm/CMakeLists.txt
@@ -39,7 +39,9 @@ if(${ANDROID})
   find_library(android_log log)
 endif()
 
-add_compile_options("-Wall" "-Werror" "-Wno-sign-compare")
+if(NOT MSVC)
+  add_compile_options("-Wall" "-Werror" "-Wno-sign-compare")
+endif()
 add_compile_definitions(C10_USING_CUSTOM_GENERATED_MACROS)
 
 # GNU emit wanring for ignored attributes Unfortunately, we use [[maybe_unused]]
@@ -51,10 +53,13 @@ endif()
 
 if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
   # strip symbols
-  add_link_options("-s")
-
-  # --gc-sections is added by torch.
-  add_compile_options("-O3" "-ffunction-sections" "-fdata-sections" "-frtti")
+  if(NOT MSVC)
+    add_link_options("-s")
+    # --gc-sections is added by torch.
+    add_compile_options("-O3" "-ffunction-sections" "-fdata-sections" "-frtti")
+  else()
+    add_compile_options("/O2" "/Gy" "/Gw" "/GR")
+  endif()
 endif()
 
 include_directories(
@@ -118,6 +123,7 @@ add_library(qnn_mem_manager STATIC)
 add_library(qnn_op_package_manager STATIC)
 add_library(qnn_profiler STATIC)
 add_library(qnn_schema INTERFACE ${_qnn_schema__outputs})
+add_dependencies(qnn_schema flatc)
 add_library(qnn_sys_function_interface INTERFACE)
 add_library(qnn_sys_implementation STATIC)
 add_library(shared_buffer STATIC)
@@ -260,9 +266,13 @@ if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64")
 
   if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
     # need to allow exceptions in pybind
-    set(_pybind_compile_options -Wno-deprecated-declarations -fPIC -frtti
-                                -fexceptions
-    )
+    if(MSVC)
+      set(_pybind_compile_options /wd4996 /GR /EHsc)
+    else()
+      set(_pybind_compile_options -Wno-deprecated-declarations -fPIC -frtti
+                                  -fexceptions
+      )
+    endif()
     target_compile_options(
       PyQnnManagerAdaptor PUBLIC ${_pybind_compile_options}
     )
diff --git a/backends/samsung/CMakeLists.txt b/backends/samsung/CMakeLists.txt
index fff3ece523..be83b7ef9b 100644
--- a/backends/samsung/CMakeLists.txt
+++ b/backends/samsung/CMakeLists.txt
@@ -26,7 +26,9 @@ if(NOT FLATC_EXECUTABLE)
   set(FLATC_EXECUTABLE flatc)
 endif()
 
-add_compile_options(-Wall -Werror -fPIC)
+if(NOT MSVC)
+  add_compile_options(-Wall -Werror -fPIC)
+endif()
 if(CMAKE_BUILD_TYPE STREQUAL "Release")
   # strip symbols
   add_link_options("-s")
@@ -144,7 +146,10 @@ if(${ANDROID})
   add_library(enn_backend STATIC)
   target_link_libraries(enn_backend PRIVATE enn_logging)
   executorch_target_link_options_shared_lib(enn_backend)
-  target_compile_options(enn_backend PRIVATE -Wno-deprecated-declarations)
+  target_compile_options(enn_backend PRIVATE
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
+  )
 
   set(__enn_executor_runner_srcs
       ${EXECUTORCH_SOURCE_DIR}/examples/samsung/executor_runner/enn_executor_runner.cpp
diff --git a/backends/vulkan/CMakeLists.txt b/backends/vulkan/CMakeLists.txt
index 29ff90e729..a847fcd1cc 100644
--- a/backends/vulkan/CMakeLists.txt
+++ b/backends/vulkan/CMakeLists.txt
@@ -97,6 +97,7 @@ add_custom_command(
 
 add_library(vulkan_schema INTERFACE ${GENERATED_HEADER})
 set_target_properties(vulkan_schema PROPERTIES LINKER_LANGUAGE CXX)
+add_dependencies(vulkan_schema flatc)
 
 target_include_directories(
   vulkan_schema
diff --git a/backends/xnnpack/CMakeLists.txt b/backends/xnnpack/CMakeLists.txt
index 33bf84b906..2b69188d37 100644
--- a/backends/xnnpack/CMakeLists.txt
+++ b/backends/xnnpack/CMakeLists.txt
@@ -35,7 +35,11 @@ if(EXECUTORCH_XNNPACK_ENABLE_KLEIDI)
   add_definitions(-DENABLE_XNNPACK_KLEIDI)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 set(_xnnpack_schema__include_dir "${CMAKE_BINARY_DIR}/schema/include")
 # Paths to headers generated from the .fbs files.
@@ -86,6 +90,7 @@ unset(MV_COMMAND)
 
 add_library(xnnpack_schema INTERFACE ${_xnnpack_schema__outputs})
 set_target_properties(xnnpack_schema PROPERTIES LINKER_LANGUAGE CXX)
+add_dependencies(xnnpack_schema flatc)
 target_include_directories(
   xnnpack_schema
   INTERFACE
diff --git a/codegen/tools/CMakeLists.txt b/codegen/tools/CMakeLists.txt
index 489a96aafb..f83607aefe 100644
--- a/codegen/tools/CMakeLists.txt
+++ b/codegen/tools/CMakeLists.txt
@@ -24,7 +24,9 @@ target_include_directories(
 
 # Compile options
 target_compile_options(
-  selective_build PUBLIC -Wno-deprecated-declarations -fPIC -frtti -fexceptions
+  selective_build
+  PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -fPIC -frtti -fexceptions -Werror -Wunused-variable -Wno-unknown-argument>
+         $<$<CXX_COMPILER_ID:MSVC>:/wd4996 /GR /EHsc /WX>
 )
 
 # Link against required libraries
diff --git a/configurations/CMakeLists.txt b/configurations/CMakeLists.txt
index fa5412ac47..a07120749a 100644
--- a/configurations/CMakeLists.txt
+++ b/configurations/CMakeLists.txt
@@ -18,7 +18,10 @@ endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
-set(_common_compile_options -Wno-deprecated-declarations)
+set(_common_compile_options
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
+)
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 include(${EXECUTORCH_ROOT}/tools/cmake/Codegen.cmake)
diff --git a/devtools/bundled_program/CMakeLists.txt b/devtools/bundled_program/CMakeLists.txt
index e9c5e0e424..adf6c9b76a 100644
--- a/devtools/bundled_program/CMakeLists.txt
+++ b/devtools/bundled_program/CMakeLists.txt
@@ -40,6 +40,7 @@ add_library(
   bundled_program ${_schema_outputs}
                   ${CMAKE_CURRENT_SOURCE_DIR}/bundled_program.cpp
 )
+add_dependencies(bundled_program flatc)
 target_link_libraries(bundled_program PUBLIC executorch)
 target_include_directories(
   bundled_program
diff --git a/devtools/etdump/CMakeLists.txt b/devtools/etdump/CMakeLists.txt
index ca4df1d2a8..5b7ac238d9 100644
--- a/devtools/etdump/CMakeLists.txt
+++ b/devtools/etdump/CMakeLists.txt
@@ -49,6 +49,7 @@ add_library(
   ${CMAKE_CURRENT_SOURCE_DIR}/data_sinks/file_data_sink.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/data_sinks/file_data_sink.h
 )
+add_dependencies(etdump flatcc_cli)
 target_link_libraries(
   etdump
   PUBLIC flatccrt
diff --git a/examples/apple/mps/CMakeLists.txt b/examples/apple/mps/CMakeLists.txt
index 8a562dd206..2717a072d2 100644
--- a/examples/apple/mps/CMakeLists.txt
+++ b/examples/apple/mps/CMakeLists.txt
@@ -29,13 +29,19 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-add_compile_options("-Wall" "-Werror")
+if(NOT MSVC)
+  add_compile_options("-Wall" "-Werror")
+endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC
-                            -DET_EVENT_TRACER_ENABLED
-)
+if(MSVC)
+  set(_common_compile_options /wd4996 -DET_EVENT_TRACER_ENABLED)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC
+                              -DET_EVENT_TRACER_ENABLED
+  )
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/devtools/CMakeLists.txt b/examples/devtools/CMakeLists.txt
index 355ff37536..09726e1ac0 100644
--- a/examples/devtools/CMakeLists.txt
+++ b/examples/devtools/CMakeLists.txt
@@ -29,7 +29,11 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/mediatek/CMakeLists.txt b/examples/mediatek/CMakeLists.txt
index f93954bff9..b03f0970d5 100644
--- a/examples/mediatek/CMakeLists.txt
+++ b/examples/mediatek/CMakeLists.txt
@@ -26,7 +26,11 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories
diff --git a/examples/models/llama/CMakeLists.txt b/examples/models/llama/CMakeLists.txt
index e7c73c0cff..4e15a16549 100644
--- a/examples/models/llama/CMakeLists.txt
+++ b/examples/models/llama/CMakeLists.txt
@@ -63,7 +63,11 @@ else()
   set(CMAKE_TOOLCHAIN_IOS OFF)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/models/llava/CMakeLists.txt b/examples/models/llava/CMakeLists.txt
index cf9d54ad3e..e0bdcf88d6 100644
--- a/examples/models/llava/CMakeLists.txt
+++ b/examples/models/llava/CMakeLists.txt
@@ -60,7 +60,11 @@ else()
   set(CMAKE_TOOLCHAIN_IOS OFF)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/portable/custom_ops/CMakeLists.txt b/examples/portable/custom_ops/CMakeLists.txt
index 4188554af7..c6c0c67398 100644
--- a/examples/portable/custom_ops/CMakeLists.txt
+++ b/examples/portable/custom_ops/CMakeLists.txt
@@ -34,7 +34,11 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/qualcomm/CMakeLists.txt b/examples/qualcomm/CMakeLists.txt
index 19190b6f79..cdbd43ed13 100644
--- a/examples/qualcomm/CMakeLists.txt
+++ b/examples/qualcomm/CMakeLists.txt
@@ -32,7 +32,11 @@ find_package(executorch CONFIG REQUIRED)
 target_compile_options(executorch INTERFACE -DET_EVENT_TRACER_ENABLED)
 find_package(gflags REQUIRED)
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 #
 # The `_<target>_srcs` lists are defined by executorch_load_build_variables.
diff --git a/examples/selective_build/advanced/CMakeLists.txt b/examples/selective_build/advanced/CMakeLists.txt
index 65ebb50bca..0233d84adf 100644
--- a/examples/selective_build/advanced/CMakeLists.txt
+++ b/examples/selective_build/advanced/CMakeLists.txt
@@ -33,9 +33,13 @@ if(NOT CMAKE_CXX_STANDARD)
   # Can't set to 11 due to executor_runner.cpp make_unique
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC
-                            -ffunction-sections -fdata-sections
-)
+if(MSVC)
+  set(_common_compile_options /wd4996 /Gy /Gw)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC
+                              -ffunction-sections -fdata-sections
+  )
+endif()
 
 add_subdirectory(${EXECUTORCH_ROOT} ${CMAKE_CURRENT_BINARY_DIR}/executorch)
 
diff --git a/examples/selective_build/basic/CMakeLists.txt b/examples/selective_build/basic/CMakeLists.txt
index 3cc68ad53b..2ad27339f3 100644
--- a/examples/selective_build/basic/CMakeLists.txt
+++ b/examples/selective_build/basic/CMakeLists.txt
@@ -33,9 +33,13 @@ if(NOT CMAKE_CXX_STANDARD)
   # Can't set to 11 due to executor_runner.cpp make_unique
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC
-                            -ffunction-sections -fdata-sections
-)
+if(MSVC)
+  set(_common_compile_options /wd4996 /Gy /Gw)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC
+                              -ffunction-sections -fdata-sections
+  )
+endif()
 
 add_subdirectory(${EXECUTORCH_ROOT} ${CMAKE_CURRENT_BINARY_DIR}/executorch)
 
diff --git a/exir/backend/test/demos/rpc/CMakeLists.txt b/exir/backend/test/demos/rpc/CMakeLists.txt
index 97f90ea9ba..0bb96128e0 100644
--- a/exir/backend/test/demos/rpc/CMakeLists.txt
+++ b/exir/backend/test/demos/rpc/CMakeLists.txt
@@ -24,7 +24,11 @@ include(${EXECUTORCH_ROOT}/tools/cmake/Test.cmake)
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 add_library(
   executor_backend STATIC ExecutorBackendRegister.cpp ExecutorBackend.cpp
diff --git a/extension/android/CMakeLists.txt b/extension/android/CMakeLists.txt
index 4755fed7bc..a6c0bfd08a 100644
--- a/extension/android/CMakeLists.txt
+++ b/extension/android/CMakeLists.txt
@@ -18,7 +18,11 @@ endif()
 
 set(EXECUTORCH_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/../..")
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 if(NOT ANDROID_PLATFORM)
   set(ANDROID_PLATFORM android-30)
 endif()
diff --git a/extension/flat_tensor/serialize/CMakeLists.txt b/extension/flat_tensor/serialize/CMakeLists.txt
index 1909bd4de0..3678639ddb 100644
--- a/extension/flat_tensor/serialize/CMakeLists.txt
+++ b/extension/flat_tensor/serialize/CMakeLists.txt
@@ -43,6 +43,7 @@ function(generate_flat_tensor_schema _schema_srcs _schema_name)
 
   add_library(${_schema_name} INTERFACE ${_schema_outputs})
   set_target_properties(${_schema_name} PROPERTIES LINKER_LANGUAGE CXX)
+  add_dependencies(${_schema_name} flatc)
 
   # exir lets users set the alignment of tensor data embedded in the flatbuffer,
   # and some users need an alignment larger than the default, which is typically
diff --git a/extension/llm/custom_ops/CMakeLists.txt b/extension/llm/custom_ops/CMakeLists.txt
index 1678dc8029..50825bee43 100644
--- a/extension/llm/custom_ops/CMakeLists.txt
+++ b/extension/llm/custom_ops/CMakeLists.txt
@@ -16,7 +16,11 @@ if(NOT EXECUTORCH_ROOT)
   set(EXECUTORCH_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../../..)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "arm64|aarch64")
   list(APPEND _common_compile_options "-march=armv8.2-a+dotprod")
 endif()
@@ -135,8 +139,10 @@ if(EXECUTORCH_BUILD_KERNELS_LLM_AOT)
     target_link_libraries(custom_ops_aot_lib PUBLIC pthreadpool cpuinfo)
   endif()
   target_compile_options(
-    custom_ops_aot_lib PUBLIC -Wno-deprecated-declarations -fPIC -frtti
-                              -fexceptions ${_common_compile_options}
+    custom_ops_aot_lib PUBLIC
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -fPIC -frtti -fexceptions>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4996 /GR /EHsc>
+    ${_common_compile_options}
   )
 
   install(
diff --git a/extension/module/CMakeLists.txt b/extension/module/CMakeLists.txt
index 5f114f1bef..79323dc512 100644
--- a/extension/module/CMakeLists.txt
+++ b/extension/module/CMakeLists.txt
@@ -35,7 +35,9 @@ target_include_directories(
   extension_module PUBLIC ${_common_include_directories}
 )
 target_compile_options(
-  extension_module PUBLIC -Wno-deprecated-declarations -fPIC
+  extension_module PUBLIC
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -fPIC>
+  $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
 )
 
 # Module extension built as a static library. TODO(gjcomer) Remove this target
@@ -49,7 +51,9 @@ target_include_directories(
   extension_module_static PUBLIC ${_common_include_directories}
 )
 target_compile_options(
-  extension_module_static PUBLIC -Wno-deprecated-declarations -fPIC
+  extension_module_static PUBLIC
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -fPIC>
+  $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
 )
 
 # Install libraries
diff --git a/extension/training/CMakeLists.txt b/extension/training/CMakeLists.txt
index 1e17913141..dc8980076f 100644
--- a/extension/training/CMakeLists.txt
+++ b/extension/training/CMakeLists.txt
@@ -70,7 +70,9 @@ if(EXECUTORCH_BUILD_PYBIND)
 
   target_include_directories(_training_lib PRIVATE ${TORCH_INCLUDE_DIRS})
   target_compile_options(
-    _training_lib PUBLIC -Wno-deprecated-declarations -fPIC -frtti -fexceptions
+    _training_lib PUBLIC
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -fPIC -frtti -fexceptions>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4996 /GR /EHsc>
   )
   target_link_libraries(_training_lib PRIVATE ${_pybind_training_dep_libs})
 
diff --git a/extension/wasm/CMakeLists.txt b/extension/wasm/CMakeLists.txt
index 36c336e17c..7ca22e31fc 100644
--- a/extension/wasm/CMakeLists.txt
+++ b/extension/wasm/CMakeLists.txt
@@ -27,7 +27,11 @@ if(NOT EXECUTORCH_ROOT)
 endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
-set(_common_compile_options -Wno-deprecated-declarations -fPIC -Wall -Werror)
+if(MSVC)
+  set(_common_compile_options /wd4996 /W4 /WX)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC -Wall -Werror)
+endif()
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
 
 set(link_libraries)
diff --git a/extension/wasm/tokenizers/CMakeLists.txt b/extension/wasm/tokenizers/CMakeLists.txt
index 03b7ea1ff6..17dad7e9f3 100644
--- a/extension/wasm/tokenizers/CMakeLists.txt
+++ b/extension/wasm/tokenizers/CMakeLists.txt
@@ -25,7 +25,11 @@ if(NOT EXECUTORCH_ROOT)
 endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
-set(_common_compile_options -Wno-deprecated-declarations -fPIC -Wall -Werror)
+if(MSVC)
+  set(_common_compile_options /wd4996 /W4 /WX)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC -Wall -Werror)
+endif()
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
 
 set(link_libraries)
diff --git a/kernels/optimized/CMakeLists.txt b/kernels/optimized/CMakeLists.txt
index 32ae865bfd..555d129b6e 100644
--- a/kernels/optimized/CMakeLists.txt
+++ b/kernels/optimized/CMakeLists.txt
@@ -21,7 +21,11 @@ if(NOT EXECUTORCH_ROOT)
   set(EXECUTORCH_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../..)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations)
+endif()
 
 # Note for apple platform we can rely on Accelerate framework Will come back to
 # this
diff --git a/kernels/portable/CMakeLists.txt b/kernels/portable/CMakeLists.txt
index eb8475b8d5..07286ef534 100644
--- a/kernels/portable/CMakeLists.txt
+++ b/kernels/portable/CMakeLists.txt
@@ -21,7 +21,11 @@ if(NOT EXECUTORCH_ROOT)
   set(EXECUTORCH_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../..)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations)
+endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 include(${EXECUTORCH_ROOT}/tools/cmake/Codegen.cmake)
diff --git a/kernels/portable/cpu/util/CMakeLists.txt b/kernels/portable/cpu/util/CMakeLists.txt
index 047760f321..c1d8d19e86 100644
--- a/kernels/portable/cpu/util/CMakeLists.txt
+++ b/kernels/portable/cpu/util/CMakeLists.txt
@@ -21,7 +21,11 @@ endif()
 
 list(TRANSFORM _kernels_util_all_deps__srcs PREPEND "${EXECUTORCH_ROOT}/")
 
-set(_common_compile_options -Wno-deprecated-declarations)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations)
+endif()
 
 add_library(kernels_util_all_deps ${_kernels_util_all_deps__srcs})
 target_link_libraries(kernels_util_all_deps PRIVATE executorch_core)
diff --git a/kernels/quantized/CMakeLists.txt b/kernels/quantized/CMakeLists.txt
index b0c837cdef..75a5e80b18 100644
--- a/kernels/quantized/CMakeLists.txt
+++ b/kernels/quantized/CMakeLists.txt
@@ -20,7 +20,10 @@ if(NOT EXECUTORCH_ROOT)
   set(EXECUTORCH_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../..)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations)
+set(_common_compile_options
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
+)
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 include(${EXECUTORCH_ROOT}/tools/cmake/Codegen.cmake)
diff --git a/schema/CMakeLists.txt b/schema/CMakeLists.txt
index fc844bbf3a..d9ac68fd45 100644
--- a/schema/CMakeLists.txt
+++ b/schema/CMakeLists.txt
@@ -39,6 +39,7 @@ function(generate_program_schema _schema_srcs _schema_name)
 
   add_library(${_schema_name} INTERFACE ${_schema_outputs})
   set_target_properties(${_schema_name} PROPERTIES LINKER_LANGUAGE CXX)
+  add_dependencies(${_schema_name} flatc)
 
   # exir lets users set the alignment of tensor data embedded in the flatbuffer,
   # and some users need an alignment larger than the default, which is typically
diff --git a/third-party/CMakeLists.txt b/third-party/CMakeLists.txt
index 767ac367e1..3586f14d27 100644
--- a/third-party/CMakeLists.txt
+++ b/third-party/CMakeLists.txt
@@ -27,6 +27,11 @@ endif()
 
 # We use ExternalProject to build flatc from source to force it target the host.
 # Otherwise, flatc will target the project's toolchain (i.e. iOS, or Android).
+if(WIN32)
+  set(_flatc_executable flatc.exe)
+else()
+  set(_flatc_executable flatc)
+endif()
 ExternalProject_Add(
   flatbuffers_ep
   PREFIX ${CMAKE_CURRENT_BINARY_DIR}/flatc_ep
@@ -44,12 +49,15 @@ ExternalProject_Add(
              # If building for iOS, "unset" these variables to rely on the host (macOS) defaults.
              $<$<AND:$<BOOL:${APPLE}>,$<BOOL:$<FILTER:${PLATFORM},EXCLUDE,^MAC>>>:-DCMAKE_OSX_SYSROOT=>
              -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${CMAKE_OSX_DEPLOYMENT_TARGET}
-  BUILD_BYPRODUCTS <INSTALL_DIR>/bin/flatc
+  BUILD_BYPRODUCTS <INSTALL_DIR>/bin/${_flatc_executable}
   ${_executorch_external_project_additional_args}
 )
+# Create explicit step targets so Ninja can properly track dependencies
+ExternalProject_Add_StepTargets(flatbuffers_ep install)
 ExternalProject_Get_Property(flatbuffers_ep INSTALL_DIR)
 add_executable(flatc IMPORTED GLOBAL)
-add_dependencies(flatc flatbuffers_ep)
+# Depend on the install step to ensure flatc.exe exists before use
+add_dependencies(flatc flatbuffers_ep-install)
 if(WIN32 AND NOT CMAKE_CROSSCOMPILING)
   # flatbuffers does not use CMAKE_BUILD_TYPE. Internally, the build forces Release
   # config, but from CMake's perspective the build type is always Debug.
@@ -79,6 +87,11 @@ endif()
 
 # Similar to flatbuffers, we want to build flatcc for the host. See inline comments
 # in the flatbuffers ExternalProject_Add for more details.
+if(WIN32)
+  set(_flatcc_executable flatcc.exe)
+else()
+  set(_flatcc_executable flatcc)
+endif()
 ExternalProject_Add(
   flatcc_ep
   PREFIX ${CMAKE_CURRENT_BINARY_DIR}/flatcc_ep
@@ -96,13 +109,16 @@ ExternalProject_Add(
              $<$<AND:$<BOOL:${APPLE}>,$<BOOL:$<FILTER:${PLATFORM},EXCLUDE,^MAC>>>:-DCMAKE_OSX_SYSROOT=>
              -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${CMAKE_OSX_DEPLOYMENT_TARGET}
              ${_flatcc_extra_cmake_args}
-  BUILD_BYPRODUCTS <INSTALL_DIR>/bin/flatcc
-  {_executorch_external_project_additional_args}
+  BUILD_BYPRODUCTS <INSTALL_DIR>/bin/${_flatcc_executable}
+  ${_executorch_external_project_additional_args}
 )
+# Create explicit step targets so Ninja can properly track dependencies
+ExternalProject_Add_StepTargets(flatcc_ep install)
 file(REMOVE_RECURSE ${PROJECT_SOURCE_DIR}/third-party/flatcc/lib)
 ExternalProject_Get_Property(flatcc_ep INSTALL_DIR)
 add_executable(flatcc_cli IMPORTED GLOBAL)
-add_dependencies(flatcc_cli flatcc_ep)
+# Depend on the install step to ensure flatcc.exe exists before use
+add_dependencies(flatcc_cli flatcc_ep-install)
 if(WIN32 AND NOT CMAKE_CROSSCOMPILING)
   set_target_properties(flatcc_cli PROPERTIES IMPORTED_LOCATION ${INSTALL_DIR}/bin/flatcc.exe)
 else()
diff --git a/tools/cmake/Codegen.cmake b/tools/cmake/Codegen.cmake
index 32d3d8b554..01e205d42f 100644
--- a/tools/cmake/Codegen.cmake
+++ b/tools/cmake/Codegen.cmake
@@ -190,7 +190,10 @@ function(gen_custom_ops_aot_lib)
   )
   find_package_torch()
   # This lib uses ATen lib, so we explicitly enable rtti and exceptions.
-  target_compile_options(${GEN_LIB_NAME} PRIVATE -frtti -fexceptions)
+  target_compile_options(${GEN_LIB_NAME} PRIVATE
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti -fexceptions>
+    $<$<CXX_COMPILER_ID:MSVC>:/GR /EHsc>
+  )
   target_compile_definitions(${GEN_LIB_NAME} PRIVATE USE_ATEN_LIB=1)
   include_directories(${TORCH_INCLUDE_DIRS})
   target_link_libraries(${GEN_LIB_NAME} PRIVATE torch)
@@ -232,8 +235,9 @@ function(gen_operators_lib)
   set(portable_kernels_check "portable_kernels")
   if(GEN_KERNEL_LIBS)
 
-    set(_common_compile_options -Wno-deprecated-declarations
-                                -ffunction-sections -fdata-sections -Os
+    set(_common_compile_options
+        $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -ffunction-sections -fdata-sections -Os>
+        $<$<CXX_COMPILER_ID:MSVC>:/wd4996 /Gy /Gw /Os>
     )
 
     if(GEN_DTYPE_SELECTIVE_BUILD)
@@ -353,6 +357,7 @@ endfunction()
 # Append the file list in the variable named `name` in build/build_variables.bzl
 # to the variable named `outputvar` in the caller's scope.
 function(executorch_append_filelist name outputvar)
+  set(_rootdir "${EXECUTORCH_ROOT}")
   # configure_file adds its input to the list of CMAKE_RERUN dependencies
   configure_file(
     ${EXECUTORCH_ROOT}/shim_et/xplat/executorch/build/build_variables.bzl
@@ -361,10 +366,12 @@ function(executorch_append_filelist name outputvar)
   if(NOT PYTHON_EXECUTABLE)
     resolve_python_executable()
   endif()
+  # Convert to forward slashes for Python compatibility on Windows
+  file(TO_CMAKE_PATH "${EXECUTORCH_ROOT}" _et_root_posix)
   execute_process(
     COMMAND
       "${PYTHON_EXECUTABLE}" -c
-      "exec(open('${EXECUTORCH_ROOT}/shim_et/xplat/executorch/build/build_variables.bzl').read());print(';'.join(${name}))"
+      "exec(open('${_et_root_posix}/shim_et/xplat/executorch/build/build_variables.bzl').read());print(';'.join(${name}))"
     WORKING_DIRECTORY "${_rootdir}"
     RESULT_VARIABLE _retval
     OUTPUT_VARIABLE _tempvar
diff --git a/tools/cmake/Utils.cmake b/tools/cmake/Utils.cmake
index 77918ebbf2..c6c09e8364 100644
--- a/tools/cmake/Utils.cmake
+++ b/tools/cmake/Utils.cmake
@@ -84,10 +84,17 @@ function(resolve_python_executable)
         PARENT_SCOPE
     )
   elseif(DEFINED ENV{VIRTUAL_ENV})
-    set(PYTHON_EXECUTABLE
-        $ENV{VIRTUAL_ENV}/bin/python3
-        PARENT_SCOPE
-    )
+    if(WIN32)
+      set(PYTHON_EXECUTABLE
+          $ENV{VIRTUAL_ENV}/Scripts/python.exe
+          PARENT_SCOPE
+      )
+    else()
+      set(PYTHON_EXECUTABLE
+          $ENV{VIRTUAL_ENV}/bin/python3
+          PARENT_SCOPE
+      )
+    endif()
   else()
     set(PYTHON_EXECUTABLE
         python3
