diff --git a/backends/apple/coreml/CMakeLists.txt b/backends/apple/coreml/CMakeLists.txt
index 9879a05e3d..a0ad095ebc 100644
--- a/backends/apple/coreml/CMakeLists.txt
+++ b/backends/apple/coreml/CMakeLists.txt
@@ -113,7 +113,7 @@ target_include_directories(
 if(APPLE)
   target_link_libraries(coreml_util PRIVATE ${FOUNDATION_FRAMEWORK})
 endif()
-target_compile_options(coreml_util PUBLIC -fPIC)
+target_compile_options(coreml_util PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-fPIC>)
 
 install(TARGETS coreml_util DESTINATION lib)
 
@@ -150,7 +150,7 @@ if(APPLE)
     coreml_inmemoryfs PRIVATE coreml_util ${FOUNDATION_FRAMEWORK}
   )
 endif()
-target_compile_options(coreml_inmemoryfs PUBLIC -fPIC)
+target_compile_options(coreml_inmemoryfs PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-fPIC>)
 
 install(
   TARGETS coreml_inmemoryfs
@@ -178,7 +178,7 @@ if(EXECUTORCH_BUILD_PYBIND)
     executorchcoreml PRIVATE coreml_util coreml_inmemoryfs
                              nlohmann_json::nlohmann_json
   )
-  target_compile_options(executorchcoreml PUBLIC -fPIC)
+  target_compile_options(executorchcoreml PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-fPIC>)
 endif()
 
 # coremldelegate
@@ -242,8 +242,8 @@ if(APPLE)
 
   if(EXECUTORCH_BUILD_DEVTOOLS)
     target_compile_options(executorch_core PUBLIC -DET_EVENT_TRACER_ENABLED)
-    target_compile_options(coremldelegate PRIVATE "-frtti")
-    target_compile_options(libprotobuf-lite PRIVATE "-frtti")
+    target_compile_options(coremldelegate PRIVATE $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti> $<$<CXX_COMPILER_ID:MSVC>:/GR>)
+    target_compile_options(libprotobuf-lite PRIVATE $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti> $<$<CXX_COMPILER_ID:MSVC>:/GR>)
   else()
     target_compile_options(coremldelegate PRIVATE "-fno-rtti")
   endif()
diff --git a/backends/apple/mps/CMakeLists.txt b/backends/apple/mps/CMakeLists.txt
index 5a253347b0..e5cfb280ec 100644
--- a/backends/apple/mps/CMakeLists.txt
+++ b/backends/apple/mps/CMakeLists.txt
@@ -18,7 +18,11 @@ endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
-set(_common_compile_options -Wno-deprecated-declarations)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations)
+endif()
 
 set(_mps_schema__include_dir "${CMAKE_BINARY_DIR}/schema/include")
 
diff --git a/backends/arm/CMakeLists.txt b/backends/arm/CMakeLists.txt
index ede7a96a38..5b65abf177 100644
--- a/backends/arm/CMakeLists.txt
+++ b/backends/arm/CMakeLists.txt
@@ -22,7 +22,9 @@ add_compile_definitions(C10_USING_CUSTOM_GENERATED_MACROS)
 # bare metal backend builds
 if(EXECUTORCH_BUILD_ARM_BAREMETAL)
 
-  add_compile_options("-Wall" "-Werror")
+  if(NOT MSVC)
+    add_compile_options("-Wall" "-Werror")
+  endif()
 
   # Third-party folder and Ethos-U driver inclued
   set(THIRD_PARTY_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/third-party")
diff --git a/backends/cadence/CMakeLists.txt b/backends/cadence/CMakeLists.txt
index 47183bed21..e73cf76af2 100644
--- a/backends/cadence/CMakeLists.txt
+++ b/backends/cadence/CMakeLists.txt
@@ -31,7 +31,11 @@ add_compile_definitions(C10_USING_CUSTOM_GENERATED_MACROS)
 if(EXECUTORCH_CADENCE_CPU_RUNNER)
   include(${EXECUTORCH_ROOT}/tools/cmake/Codegen.cmake)
 
-  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+  if(MSVC)
+    set(_common_compile_options /wd4996)
+  else()
+    set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+  endif()
 
   # Find prebuilt libraries. executorch package should contain portable_ops_lib,
   # etdump, bundled_program.
diff --git a/backends/mediatek/CMakeLists.txt b/backends/mediatek/CMakeLists.txt
index ed9b37e199..a24aa3680e 100644
--- a/backends/mediatek/CMakeLists.txt
+++ b/backends/mediatek/CMakeLists.txt
@@ -23,7 +23,10 @@ include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/runtime/include)
 
 # targets
 add_library(neuron_backend SHARED)
-target_compile_options(neuron_backend PRIVATE "-frtti" "-fexceptions")
+target_compile_options(neuron_backend PRIVATE
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti -fexceptions>
+  $<$<CXX_COMPILER_ID:MSVC>:/GR /EHsc>
+)
 target_link_libraries(
   neuron_backend PRIVATE executorch_core portable_ops_lib portable_kernels
                          android log
diff --git a/backends/openvino/CMakeLists.txt b/backends/openvino/CMakeLists.txt
index 4d32d8932c..5f86624902 100644
--- a/backends/openvino/CMakeLists.txt
+++ b/backends/openvino/CMakeLists.txt
@@ -35,7 +35,10 @@ find_package(OpenVINO REQUIRED)
 add_library(openvino_backend STATIC .)
 
 # Enable exceptions and RTTI for OpenVINO backend
-target_compile_options(openvino_backend PRIVATE -frtti -fexceptions)
+target_compile_options(openvino_backend PRIVATE
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti -fexceptions>
+  $<$<CXX_COMPILER_ID:MSVC>:/GR /EHsc>
+)
 
 # Include Executorch directories
 target_include_directories(openvino_backend PRIVATE ${COMMON_INCLUDE_DIRS})
diff --git a/backends/qualcomm/CMakeLists.txt b/backends/qualcomm/CMakeLists.txt
index 3210559726..6b74f18907 100644
--- a/backends/qualcomm/CMakeLists.txt
+++ b/backends/qualcomm/CMakeLists.txt
@@ -39,7 +39,9 @@ if(${ANDROID})
   find_library(android_log log)
 endif()
 
-add_compile_options("-Wall" "-Werror" "-Wno-sign-compare")
+if(NOT MSVC)
+  add_compile_options("-Wall" "-Werror" "-Wno-sign-compare")
+endif()
 add_compile_definitions(C10_USING_CUSTOM_GENERATED_MACROS)
 
 # GNU emit wanring for ignored attributes Unfortunately, we use [[maybe_unused]]
@@ -51,10 +53,13 @@ endif()
 
 if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
   # strip symbols
-  add_link_options("-s")
-
-  # --gc-sections is added by torch.
-  add_compile_options("-O3" "-ffunction-sections" "-fdata-sections" "-frtti")
+  if(NOT MSVC)
+    add_link_options("-s")
+    # --gc-sections is added by torch.
+    add_compile_options("-O3" "-ffunction-sections" "-fdata-sections" "-frtti")
+  else()
+    add_compile_options("/O2" "/Gy" "/Gw" "/GR")
+  endif()
 endif()
 
 include_directories(
diff --git a/backends/samsung/CMakeLists.txt b/backends/samsung/CMakeLists.txt
index fff3ece523..50b13d4fa6 100644
--- a/backends/samsung/CMakeLists.txt
+++ b/backends/samsung/CMakeLists.txt
@@ -26,7 +26,9 @@ if(NOT FLATC_EXECUTABLE)
   set(FLATC_EXECUTABLE flatc)
 endif()
 
-add_compile_options(-Wall -Werror -fPIC)
+if(NOT MSVC)
+  add_compile_options(-Wall -Werror -fPIC)
+endif()
 if(CMAKE_BUILD_TYPE STREQUAL "Release")
   # strip symbols
   add_link_options("-s")
diff --git a/codegen/tools/CMakeLists.txt b/codegen/tools/CMakeLists.txt
index 489a96aafb..f83607aefe 100644
--- a/codegen/tools/CMakeLists.txt
+++ b/codegen/tools/CMakeLists.txt
@@ -24,7 +24,9 @@ target_include_directories(
 
 # Compile options
 target_compile_options(
-  selective_build PUBLIC -Wno-deprecated-declarations -fPIC -frtti -fexceptions
+  selective_build
+  PUBLIC $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -fPIC -frtti -fexceptions -Werror -Wunused-variable -Wno-unknown-argument>
+         $<$<CXX_COMPILER_ID:MSVC>:/wd4996 /GR /EHsc /WX>
 )
 
 # Link against required libraries
diff --git a/configurations/CMakeLists.txt b/configurations/CMakeLists.txt
index fa5412ac47..a07120749a 100644
--- a/configurations/CMakeLists.txt
+++ b/configurations/CMakeLists.txt
@@ -18,7 +18,10 @@ endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
-set(_common_compile_options -Wno-deprecated-declarations)
+set(_common_compile_options
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
+)
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 include(${EXECUTORCH_ROOT}/tools/cmake/Codegen.cmake)
diff --git a/examples/apple/mps/CMakeLists.txt b/examples/apple/mps/CMakeLists.txt
index 8a562dd206..2717a072d2 100644
--- a/examples/apple/mps/CMakeLists.txt
+++ b/examples/apple/mps/CMakeLists.txt
@@ -29,13 +29,19 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-add_compile_options("-Wall" "-Werror")
+if(NOT MSVC)
+  add_compile_options("-Wall" "-Werror")
+endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC
-                            -DET_EVENT_TRACER_ENABLED
-)
+if(MSVC)
+  set(_common_compile_options /wd4996 -DET_EVENT_TRACER_ENABLED)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC
+                              -DET_EVENT_TRACER_ENABLED
+  )
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/devtools/CMakeLists.txt b/examples/devtools/CMakeLists.txt
index 355ff37536..09726e1ac0 100644
--- a/examples/devtools/CMakeLists.txt
+++ b/examples/devtools/CMakeLists.txt
@@ -29,7 +29,11 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/mediatek/CMakeLists.txt b/examples/mediatek/CMakeLists.txt
index f93954bff9..b03f0970d5 100644
--- a/examples/mediatek/CMakeLists.txt
+++ b/examples/mediatek/CMakeLists.txt
@@ -26,7 +26,11 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories
diff --git a/examples/models/llama/CMakeLists.txt b/examples/models/llama/CMakeLists.txt
index e7c73c0cff..4e15a16549 100644
--- a/examples/models/llama/CMakeLists.txt
+++ b/examples/models/llama/CMakeLists.txt
@@ -63,7 +63,11 @@ else()
   set(CMAKE_TOOLCHAIN_IOS OFF)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/models/llava/CMakeLists.txt b/examples/models/llava/CMakeLists.txt
index cf9d54ad3e..e0bdcf88d6 100644
--- a/examples/models/llava/CMakeLists.txt
+++ b/examples/models/llava/CMakeLists.txt
@@ -60,7 +60,11 @@ else()
   set(CMAKE_TOOLCHAIN_IOS OFF)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/portable/custom_ops/CMakeLists.txt b/examples/portable/custom_ops/CMakeLists.txt
index 4188554af7..c6c0c67398 100644
--- a/examples/portable/custom_ops/CMakeLists.txt
+++ b/examples/portable/custom_ops/CMakeLists.txt
@@ -34,7 +34,11 @@ if(NOT PYTHON_EXECUTABLE)
   resolve_python_executable()
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 # Let files say "include <executorch/path/to/header.h>".
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
diff --git a/examples/qualcomm/CMakeLists.txt b/examples/qualcomm/CMakeLists.txt
index 19190b6f79..cdbd43ed13 100644
--- a/examples/qualcomm/CMakeLists.txt
+++ b/examples/qualcomm/CMakeLists.txt
@@ -32,7 +32,11 @@ find_package(executorch CONFIG REQUIRED)
 target_compile_options(executorch INTERFACE -DET_EVENT_TRACER_ENABLED)
 find_package(gflags REQUIRED)
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 #
 # The `_<target>_srcs` lists are defined by executorch_load_build_variables.
diff --git a/examples/selective_build/advanced/CMakeLists.txt b/examples/selective_build/advanced/CMakeLists.txt
index 65ebb50bca..0233d84adf 100644
--- a/examples/selective_build/advanced/CMakeLists.txt
+++ b/examples/selective_build/advanced/CMakeLists.txt
@@ -33,9 +33,13 @@ if(NOT CMAKE_CXX_STANDARD)
   # Can't set to 11 due to executor_runner.cpp make_unique
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC
-                            -ffunction-sections -fdata-sections
-)
+if(MSVC)
+  set(_common_compile_options /wd4996 /Gy /Gw)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC
+                              -ffunction-sections -fdata-sections
+  )
+endif()
 
 add_subdirectory(${EXECUTORCH_ROOT} ${CMAKE_CURRENT_BINARY_DIR}/executorch)
 
diff --git a/examples/selective_build/basic/CMakeLists.txt b/examples/selective_build/basic/CMakeLists.txt
index 3cc68ad53b..2ad27339f3 100644
--- a/examples/selective_build/basic/CMakeLists.txt
+++ b/examples/selective_build/basic/CMakeLists.txt
@@ -33,9 +33,13 @@ if(NOT CMAKE_CXX_STANDARD)
   # Can't set to 11 due to executor_runner.cpp make_unique
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations -fPIC
-                            -ffunction-sections -fdata-sections
-)
+if(MSVC)
+  set(_common_compile_options /wd4996 /Gy /Gw)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC
+                              -ffunction-sections -fdata-sections
+  )
+endif()
 
 add_subdirectory(${EXECUTORCH_ROOT} ${CMAKE_CURRENT_BINARY_DIR}/executorch)
 
diff --git a/exir/backend/test/demos/rpc/CMakeLists.txt b/exir/backend/test/demos/rpc/CMakeLists.txt
index 97f90ea9ba..0bb96128e0 100644
--- a/exir/backend/test/demos/rpc/CMakeLists.txt
+++ b/exir/backend/test/demos/rpc/CMakeLists.txt
@@ -24,7 +24,11 @@ include(${EXECUTORCH_ROOT}/tools/cmake/Test.cmake)
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
-set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+if(MSVC)
+  set(_common_compile_options /wd4996)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+endif()
 
 add_library(
   executor_backend STATIC ExecutorBackendRegister.cpp ExecutorBackend.cpp
diff --git a/extension/wasm/tokenizers/CMakeLists.txt b/extension/wasm/tokenizers/CMakeLists.txt
index 03b7ea1ff6..17dad7e9f3 100644
--- a/extension/wasm/tokenizers/CMakeLists.txt
+++ b/extension/wasm/tokenizers/CMakeLists.txt
@@ -25,7 +25,11 @@ if(NOT EXECUTORCH_ROOT)
 endif()
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
-set(_common_compile_options -Wno-deprecated-declarations -fPIC -Wall -Werror)
+if(MSVC)
+  set(_common_compile_options /wd4996 /W4 /WX)
+else()
+  set(_common_compile_options -Wno-deprecated-declarations -fPIC -Wall -Werror)
+endif()
 set(_common_include_directories ${EXECUTORCH_ROOT}/..)
 
 set(link_libraries)
diff --git a/kernels/quantized/CMakeLists.txt b/kernels/quantized/CMakeLists.txt
index b0c837cdef..75a5e80b18 100644
--- a/kernels/quantized/CMakeLists.txt
+++ b/kernels/quantized/CMakeLists.txt
@@ -20,7 +20,10 @@ if(NOT EXECUTORCH_ROOT)
   set(EXECUTORCH_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../..)
 endif()
 
-set(_common_compile_options -Wno-deprecated-declarations)
+set(_common_compile_options
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4996>
+)
 
 include(${EXECUTORCH_ROOT}/tools/cmake/Utils.cmake)
 include(${EXECUTORCH_ROOT}/tools/cmake/Codegen.cmake)
diff --git a/tools/cmake/Codegen.cmake b/tools/cmake/Codegen.cmake
index 32d3d8b554..01e205d42f 100644
--- a/tools/cmake/Codegen.cmake
+++ b/tools/cmake/Codegen.cmake
@@ -190,7 +190,10 @@ function(gen_custom_ops_aot_lib)
   )
   find_package_torch()
   # This lib uses ATen lib, so we explicitly enable rtti and exceptions.
-  target_compile_options(${GEN_LIB_NAME} PRIVATE -frtti -fexceptions)
+  target_compile_options(${GEN_LIB_NAME} PRIVATE
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-frtti -fexceptions>
+    $<$<CXX_COMPILER_ID:MSVC>:/GR /EHsc>
+  )
   target_compile_definitions(${GEN_LIB_NAME} PRIVATE USE_ATEN_LIB=1)
   include_directories(${TORCH_INCLUDE_DIRS})
   target_link_libraries(${GEN_LIB_NAME} PRIVATE torch)
@@ -232,8 +235,9 @@ function(gen_operators_lib)
   set(portable_kernels_check "portable_kernels")
   if(GEN_KERNEL_LIBS)
 
-    set(_common_compile_options -Wno-deprecated-declarations
-                                -ffunction-sections -fdata-sections -Os
+    set(_common_compile_options
+        $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated-declarations -ffunction-sections -fdata-sections -Os>
+        $<$<CXX_COMPILER_ID:MSVC>:/wd4996 /Gy /Gw /Os>
     )
 
     if(GEN_DTYPE_SELECTIVE_BUILD)
@@ -353,6 +357,7 @@ endfunction()
 # Append the file list in the variable named `name` in build/build_variables.bzl
 # to the variable named `outputvar` in the caller's scope.
 function(executorch_append_filelist name outputvar)
+  set(_rootdir "${EXECUTORCH_ROOT}")
   # configure_file adds its input to the list of CMAKE_RERUN dependencies
   configure_file(
     ${EXECUTORCH_ROOT}/shim_et/xplat/executorch/build/build_variables.bzl
@@ -361,10 +366,12 @@ function(executorch_append_filelist name outputvar)
   if(NOT PYTHON_EXECUTABLE)
     resolve_python_executable()
   endif()
+  # Convert to forward slashes for Python compatibility on Windows
+  file(TO_CMAKE_PATH "${EXECUTORCH_ROOT}" _et_root_posix)
   execute_process(
     COMMAND
       "${PYTHON_EXECUTABLE}" -c
-      "exec(open('${EXECUTORCH_ROOT}/shim_et/xplat/executorch/build/build_variables.bzl').read());print(';'.join(${name}))"
+      "exec(open('${_et_root_posix}/shim_et/xplat/executorch/build/build_variables.bzl').read());print(';'.join(${name}))"
     WORKING_DIRECTORY "${_rootdir}"
     RESULT_VARIABLE _retval
     OUTPUT_VARIABLE _tempvar
diff --git a/tools/cmake/Utils.cmake b/tools/cmake/Utils.cmake
index 77918ebbf2..c6c09e8364 100644
--- a/tools/cmake/Utils.cmake
+++ b/tools/cmake/Utils.cmake
@@ -84,10 +84,17 @@ function(resolve_python_executable)
         PARENT_SCOPE
     )
   elseif(DEFINED ENV{VIRTUAL_ENV})
-    set(PYTHON_EXECUTABLE
-        $ENV{VIRTUAL_ENV}/bin/python3
-        PARENT_SCOPE
-    )
+    if(WIN32)
+      set(PYTHON_EXECUTABLE
+          $ENV{VIRTUAL_ENV}/Scripts/python.exe
+          PARENT_SCOPE
+      )
+    else()
+      set(PYTHON_EXECUTABLE
+          $ENV{VIRTUAL_ENV}/bin/python3
+          PARENT_SCOPE
+      )
+    endif()
   else()
     set(PYTHON_EXECUTABLE
         python3
